comment/managers/followers.py

```diff
@@ -0,0 +1,47 @@
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+
+from comment.utils import get_username_for_comment
+
+
+class FollowerManager(models.Manager):
+    def is_following(self, email, model_object):
+        content_type = ContentType.objects.get_for_model(model_object)
+        return self.filter(email=email, object_id=model_object.id, content_type=content_type).exists()
+
+    def follow(self, email, username, model_object):
+        if not email or self.is_following(email, model_object):
+            return None
+        return self.create(email=email, username=username, content_object=model_object)
+
+    def unfollow(self, email, model_object):
+        content_type = ContentType.objects.get_for_model(model_object)
+        self.filter(email=email, object_id=model_object.id, content_type=content_type).delete()
+
+    def toggle_follow(self, email, username, model_object):
+        if not email:
+            return False
+        if self.is_following(email, model_object):
+            self.unfollow(email, model_object)
+            return False
+        self.follow(email, username, model_object)
+        return True
+
+    def follow_parent_thread_for_comment(self, comment):
+        """This method is used to set the comment's creator as a follower of own comment and the parent thread"""
+        if not comment.email:
+            return
+        model_object = comment
+        if not comment.is_parent:
+            model_object = comment.parent
+        username = get_username_for_comment(comment)
+        self.follow(comment.email, username, model_object)
+        if comment.is_parent:
+            self.follow(comment.email, username, comment.content_object)
+
+    def filter_for_model_object(self, model_obj):
+        content_type = ContentType.objects.get_for_model(model_obj)
+        return self.filter(content_type=content_type, object_id=model_obj.id)
+
+    def get_emails_for_model_object(self, model_obj):
+        return list(map(lambda x: x.email, self.filter_for_model_object(model_obj)))
```

Comment:
I'm not sure this is the most optimized query for the database when using `querysets`. As of now this will result in a lot of queries to the database. You can probably do this in one query with this
```python
return self.filter_for_model_object(model_obj).prefetch_related('user').values_list('user__email', flat=True)
```

Reply:
also, when we use keeping the variable name short to `model_obj`, I think we can keep the same for function names as well. In essence what I'm trying to refer here is changing the function name to `self.filter_for_model_obj`, will maybe help us save some of our time from typing.

Reply:
Since we are going to use `EMAIL_FIELD` instead of email, I believe we will need the current usage.
```python
return list(map(lambda x: x.EMAIL_FIELD, self.filter_for_model_object(model_obj)))
```
I am not sure how this can be done with string chain `user__email`

Reply:
We can just replace the hard coded `email` string with the one obtained from `model.EMAIL_FIELD`.

```python
from django.contrib.auth import get_user_model

User = get_user_model()
email_field = User.EMAIL_FIELD

emails = self.filter_for_model_object(model_obj).prefetch_related('user').values_list(f'user__{email_field}', flat=True)
```
Is there an issue with this? Even if this didn't work for some reason, there are a couple of other ways to perform this query in a more optimized manner than the one currently written.

Also, the current state is not only the most optimized form, but it doesn't actually return a `queryset` too.

Reply:
Cool. I am not sure if this approach works! but why do you think that your query is more optimised?

Reply:
> I am not sure if this approach works!
It works i have tested it on my machine.

> why do you think that your query is more optimised?

This was a mistake actually on my part. I actually thought you were not using `email` field on the model and this query would make a hit to the database for obtaining email from the user table for each value in the list(I know even this presumption was wrong). 

Now, that I see the code with my eyes open and brain working(hopefully), the query seems correct, but you should just use the `values_list` part from my suggestion to make this a `queryset`.